### PR TITLE
Makes shaking smoother, makes close range explosions more shakey 

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Makes shake_camera use more then instant steps for its animates. The concept is to not respect duration quite so strictly. So instead of instantly moving, we get the distance to travel, pick the lower of the two numbers to preserve some jerk, and multiply it by tile size * 0.7. So effectively 0.7 ticks per tile. 

I've found this works reasonably well for making shaking events look less shit, though there is room to make it a preference option in future, since this number could go between 0 for the old behavior, and say 5 for slow but not visually hazardous speeds.

Here's going from 0.7 to 0 with small and large explosions, including the explosion changes.

https://user-images.githubusercontent.com/58055496/126101979-5b6a4ee5-6083-46c7-a833-202bdb83f178.mp4

### Explosion changes
The old system for calculating explosion shake was sqrt(delta/10) when the delta could really only ever go up to like 20. So it maxed out at 1.3 or so, which doesn't really fit being hit directly by a bomb. I hate old coders.

See https://www.desmos.com/calculator/qy9zurj5gt

I'm changing the formula to sqrt(range/(dist + 1)), which works out to roughly 4.5 at 20 near_distance, which is a maxcap. Scales reasonably fast, 4 tiles or so, so this primarily effects getting hit full on with a big
bomb

## Why It's Good For The Game

Shaking should be smooth god damn it. Why the hell was this instant

Close range explosions should shake your screen more then long range ones.  <https://github.com/tgstation/tgstation/pull/34023>

## Changelog
:cl:
add: Getting hit point blank by an explosion will shake your screen a good bit more then before
qol: "Shakes" from explosions or hitting things really hard will look a bit smoother now
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
@Helianthus00 I know you made your shake changes with accessibility concerns in mind, let me know if this fucks something up please

**Comment by Helianthus00** [comment]
> @Helianthus00 I know you made your shake changes with accessibility concerns in mind, let me know if this fucks something up please

This looks great! The smoother explosions are much less harsh and it seems like they work out better for the epilepsy, at least for the person I talked to. I appreciate you looking further into this.

**Comment by LemonInTheDark** [comment]
That's great to hear, thank you

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -214,7 +214,10 @@
 		. += letter
 	return sanitize(.)
 
+#define TILES_PER_SECOND 0.7
```
Comment (Mothblocks) [inline code/modules/mob/mob_helpers.dm:217]: undef this

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -224,12 +227,31 @@
...
+	var/last_y = oldy
+
+	var/time_spent = 0
+	while(time_spent < duration)
+		//Get a random pos in our box
+		var/x_pos = rand(min, max) + oldx
+		var/y_pos = rand(min, max) + oldy
+
+		//We take the smaller of our two distances so things still have the propencity to feel somewhat jerky
+		var/time = round(max(min(abs(last_x - x_pos), abs(last_y - y_pos)) * time_scalar, 1))
```
Comment (Mothblocks) [inline code/modules/mob/mob_helpers.dm:242]: are you sure you dont want clamp? this is hard to read

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_helpers.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/mob/mob_helpers.dm:242]: I remember someone mentioning fucked behavior with clamp in the vein of like, the smallest arg is taken as the min or something. I can look deeper into it if you'd like

Comment (Mothblocks) [reply code/modules/mob/mob_helpers.dm:242]: weird, i'd've just expected it to error

Comment (LemonInTheDark) [reply code/modules/mob/mob_helpers.dm:242]: scratch that, clamp would be ok here

Comment (LemonInTheDark) [reply code/modules/mob/mob_helpers.dm:242]: oh actually no it'd make applying the scalar a headache, leaving it as is if that's ok

**Pull request merged**